src/event/ngx_event_openssl.c

```diff
@@ -4215,13 +4229,17 @@ ngx_openssl_engine(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
         return NGX_CONF_ERROR;
     }
 
+    if (!ENGINE_init(engine)) {
+        ngx_ssl_error(NGX_LOG_EMERG, cf->log, 0, "ENGINE_init(engine) failed");
+        ENGINE_free(engine);
+        return NGX_CONF_ERROR;
+    }
+
     if (ENGINE_set_default(engine, ENGINE_METHOD_ALL) == 0) {
         ngx_ssl_error(NGX_LOG_EMERG, cf->log, 0,
                       "ENGINE_set_default(\"%V\", ENGINE_METHOD_ALL) failed",
                       &value[1]);
```

Comment:
Here, the error is reported in different way than you added in the above case. Here we print `&value[1]` as `&V`, which says us which engine failed to initialize/set_default, which might be very helpful for debugging. I would consider using the same also in the other errors.

Reply:
Yes, I agree. As I commented before, the error messages should give as more information as possible. In this case the author of this part "made a mistake" not following the style of the error messages. I can follow this and "do the mistake" too and make the error messages more informative (but out of the coding style).

Reply:
Ok, I went through the code and it seems that the error messaging do not follow a strict rule. I think I can improve the messages in the parts I modified.